src/set_partner_key.c

```diff
@@ -73,11 +73,14 @@ int set_partner_key(const command_t *cmd) {
         return reply_error(err);
     }
 
-    // The incoming partner name is NOT NULL terminated, so we use memcpy and
-    // manually NULL terminate the buffer
-    memset(G_swap_ctx.partner.name, 0, sizeof(G_swap_ctx.partner.name));
```

Comment:
name and prefix are part of union. Any perticular reason you have combined to just one char array? I dont see any benefit in terms of size reduction. Union is better in terms of making it clear for what we are using the array?

Reply:
Same as above